source/common/router/router.cc

```diff
@@ -520,6 +520,21 @@ void Filter::onUpstreamHeaders(Http::HeaderMapPtr&& headers, bool end_stream) {
       upstream_request_->upstream_host_->canary();
   chargeUpstreamCode(*headers, upstream_request_->upstream_host_);
 
+  // Append routing cookies
+  for (auto& cookie : downstream_set_cookies_) {
+    const std::string& key = std::get<0>(cookie);
+    const std::string& value = std::get<1>(cookie);
+    const int max_age = std::get<2>(cookie);
+
+    // Don't overwrite a cookie if it's already set by the upstream host
+    if (Http::Utility::hasSetCookie(*headers, key)) {
```

Comment:
Do we think there's a need for this yet?  It seems unlikely that upstream can set a cookie which will hash back to the "correct" server.  I suppose we could check to make sure we're replacing - I'm not sure what browsers do if they get two set-cookies with the same key in the same response...

Reply:
Fair. I guess the only thing this actually does is allow the upstream host to throw the downstream client to another random server since it can't really even pick which one. Chrome picks the second value if the same cookie is set twice, and I'd guess the last one generally.